app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationsBoundaryCallback.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.keylesspalace.tusky.components.conversation
```

Comment:
not sure what license header to use when its only partially copied?

Reply:
Probably include both? If it's possible to refactor the code to isolate the included bits into a dedicated file / module then it's a bit easier.